app/src/org/commcare/tasks/FormRecordLoaderTask.java

```diff
@@ -151,6 +152,9 @@ protected Integer doInBackground(FormRecord... params) {
         recordTextDesc.add(DateUtils.formatDateTime(context, current.lastModified().getTime(), DateUtils.FORMAT_NO_MONTH_DAY | DateUtils.FORMAT_NO_YEAR).toLowerCase());
 
         String dataTitle = loadDataTitle(current.getID());
+        if (TextUtils.isEmpty(dataTitle) && current.getDescriptor() != null) {

```

Comment:
I think we want to preferentially use the descriptor if it is available over Data Title.

Data Title takes **quite a while to load**. If we could load the descriptor directly it would probably let us get that data in real time, but also just in general since the descriptor is user configurable we want it to "Win" over any default or implicit behavior